src/Symfony/Bundle/SecurityBundle/Form/SecurityPasswordType.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Form;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Event\SubmitEvent;
+use Symfony\Component\Form\Exception\InvalidConfigurationException;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @author Loïck Piera <pyrech@gmail.com>
+ */
+class SecurityPasswordType extends AbstractType
+{
+    private $passwordEncoder;
+
+    public function __construct(UserPasswordEncoderInterface $passwordEncoder)
+    {
+        $this->passwordEncoder = $passwordEncoder;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addEventListener(FormEvents::SUBMIT, function (SubmitEvent $event) {
+            $securityUser = $event->getForm()->getConfig()->getOption('security_user');
+
+            if (!$securityUser) {
+                $parentData = $event->getForm()->getParent()->getData();
+
+                if (!$parentData instanceof UserInterface) {
+                    throw new InvalidConfigurationException(sprintf('You should either use "%s" inside a parent form where data is an instance of "%s" or specify the user in "security_user" option', self::class, UserInterface::class));
+                }
+
+                $securityUser = $parentData;
+            }
+
+            $plainPassword = $event->getData();
+
+            $event->setData($plainPassword ? $this->passwordEncoder->encodePassword($securityUser, $plainPassword) : $securityUser->getPassword());
+        });
+    }
+
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars['required'] = false;
```

Comment:
I would remove this. If this type is e.g. used in a registration form, I can imagine the field being required. I would leave this to the user of this type.

Reply:
Yep, I don't like it so much neither. I would have preferred to override the option to just make it false by default, because it felt strange to me to require the password on edition. But could not find a way to override the default :confused:

Reply:
You can implement the `public function configureOptions(OptionsResolver $resolver)` method and call `$resolver->setDefault('required', false)` to set the default to `false`.

I'm not sure however if this is expected. I would keep the behavior the same as in `PasswordType`. If a user wants to make this optional, they can always use this as `->add('password', SecurityPasswordType::class, ['required' => false])`

Reply:
I tried to configure the default option but it did not work (maybe due to the inheritance of form, or another cause I don't understand 😅

But anyway, let's just remove it :+1: